actors/init/src/main/java/org/sunbird/init/actors/SystemInitActor.java

```diff
@@ -0,0 +1,226 @@
+      req.put(JsonKey.SLUG, slug);
+    } else {
+      ProjectCommonException.throwClientErrorException(
+          ResponseCode.slugIsNotUnique, ResponseCode.slugIsNotUnique.getErrorMessage());
+    }
+
+    boolean isChannelRegistered = Util.registerChannel(req);
+    if (!isChannelRegistered) {
+      ProjectCommonException.throwServerErrorException(
+          ResponseCode.channelRegFailed, ResponseCode.channelRegFailed.getErrorMessage());
+    }
+
+    Organization org = mapper.convertValue(req, Organization.class);
+    req = mapper.convertValue(org, Map.class);
+    Response result =
+        cassandraOperation.insertRecord(orgDbInfo.getKeySpace(), orgDbInfo.getTableName(), req);
+    addInitialisationFlagToSystemSettings();
+    ProjectLogger.log("Org data saved into cassandra.");
+    ProjectLogger.log("Created org id is ----." + uniqueId);
+    result.getResult().put(JsonKey.ORGANISATION_ID, uniqueId);
+    sender().tell(result, self());
+    Request orgReq = new Request();
+    orgReq.getRequest().put(JsonKey.ORGANISATION, req);
+    orgReq.setOperation(ActorOperations.INSERT_ORG_INFO_ELASTIC.getValue());
+    ProjectLogger.log("Calling background job to save org data into ES" + uniqueId);
+    tellToAnother(orgReq);
+  }
+
+  /**
+   * This method checks if the given channel slug value is unique
+   *
+   * @param slug input slug value need to checked.
+   * @return returns true if slug unique or return false
+   */
+  private boolean isSlugUnique(String slug) {
+    if (StringUtils.isNotBlank(slug)) {
+      Map<String, Object> filters = new HashMap<>();
+      filters.put(JsonKey.SLUG, slug);
+      filters.put(JsonKey.IS_ROOT_ORG, true);
+      Map<String, Object> esResult =
+          elasticSearchComplexSearch(
+              filters, EsIndex.sunbird.getIndexName(), EsType.organisation.getTypeName());
+      if (isNotNull(esResult)
+          && esResult.containsKey(JsonKey.CONTENT)
+          && isNotNull(esResult.get(JsonKey.CONTENT))) {
+        return (((List) esResult.get(JsonKey.CONTENT)).isEmpty());
+      }
+    }
+    return false;
+  }
+
+  /** This method writes the rootOrgInitialised status(true) to system settings */
+  private void addInitialisationFlagToSystemSettings() {
+    try {
+      this.systemSetting =
+          new SystemSetting(
+              JsonKey.IS_ROOT_ORG_INITIALISED,
+              JsonKey.IS_ROOT_ORG_INITIALISED,
+              String.valueOf(true));
+      Response response = systemSettingService.writeSetting(systemSetting);
+      ProjectLogger.log(
+          "Insert operation result for initialised status =  "
+              + response.getResult().get(JsonKey.RESPONSE),
+          LoggerEnum.DEBUG.name());
+    } catch (Exception e) {
+      ProjectLogger.log(e.getMessage(), e);
+    }
+  }
+
+  /**
+   * If system is already initalised i.e if isRootOrgInitialised of system-settings is true then
+   * throws exception
+   */
+  private void throwExceptionIfRootOrgAlreadyInitialised() {
```

Comment:
Layer this method into two methods, one which checks for initialisation and one which throws the exception.

Reply:
Suggested rename ```validateRootOrgUninitialised```